src/kaleidoscope/Kaleidoscope.cpp

```diff
@@ -34,7 +34,7 @@ Kaleidoscope_::setup(void) {
   //
   // TODO(anyone): Figure out a way we can get rid of this, and fix the bug
   // properly.
-  Serial.begin(9600);
+  KeyboardHardware.focus_interface.begin(9600);
```

Comment:
Like my comments about EEPROM, can we lift the Serial to KeyboardHardware.focus_interface change out to a small PR we can merge easily? (Even if KeyboardHardware.focus_interface just returns Serial() in the PR)

Reply:
> As a first step, can we get all of the EEPROM to KeyboardHardware.storage() changes outside of the hardware directory out of this PR by adding KeyboardHardware.storage() in the base class as a facade around EEPROM (temporarily) and pulling that all together as a small PR? It doesn't actually make things much more portable, but it helps make this PR cleaner and clearer.

I'd say... yes and no. Yes, we can get it out to a separate PR, but no, as in I wouldn't want to add the temporary facade first. Bootloaders, the Storage abstraction and the Focus interface changes are all separate commits, based on top of the former hardware API changes. I can pull those out from this PR, and have them as followup PRs, no problem. But I'd rather not add a temporary facade as a preparation step, don't really see the point.

Reply:
Originally, I wanted to turn things around, and keep the Serial and EEPROM calls in this PR, and once I cleaned things up, chop them off, and submit them as followup PRs instead of merging a small one before this. I may still need to do that with EEPROM, but for Serial, I can do the small PR route for sure.

Reply:
This has now been done as of #701, which introduces dummy `serialPort()` and `storage()` methods to `KeyboardHardware`. Once #701 is merged, the rest of this PR will be considerably smaller.

(It's been rebased on top of #701)